pyramid_of_numbers.c

```diff
@@ -0,0 +1,232 @@
+*** @param factor The int value which is multiplied by BigInt.
+*** @param *big_result The result of the multiplication.
+*/
+void multiply(const struct BigInt *big_int, int factor, struct BigInt *big_result);
+
+/** divide() multiplies a BigInt by an int.
+*** @param big_int The BigInt to be divided.
+*** @param divisor The int value by which we want to devide big_int.
+*** @param *big_result The result of the division.
+*/
+void divide(const struct BigInt *big_int, int divisor, struct BigInt *big_result);
+
+/** copy_big_int() copies a BigInt to another BigInt.
+*** @param from The source where we want to copy from.
+*** @param *to The target where we want to copy to.
+*/
+void copy_big_int(const struct BigInt *from, struct BigInt *to);
+
+/**
+*** main() reads the base number from which the pyramid has to be calculated
+*** into an array of char. The max. length of this number is MAX_DIGITS.
+*** The number is checked to contain only digits. If not the program exits.
+*** Then the inputted number is converted into a big int by calling the
+*** function strtobig_int().
+*** After the conversion the tower is calculated by calling the functions
+*** multiply(), print_big_int(), and copy_big_int() consecutively from 2 to
+*** 9 and then again with divide() instead of multiply() from 2 to 9.
+***
+*/
+int main(int argc, char *argv[])
+{
+	printf("Pyramid of Numbers\n\n");
+	char input[MAX_DIGITS];
+	printf("Please enter one number: ");
+	scanf("%s", input);
+	int length = strlen(input);
+	if(length > MAX_DIGITS){
+		printf("Error code -1\n");
+		return 0;
+	}
+
+	struct BigInt input_number = {length};
+ 	int actual = strtobig_int(input, length, &input_number);
+
+	if(actual != length){
+		printf("Error code -2\n");
+		return 0;
+	}
+
+	for (int i = 2; i <= 9; i++) {
+		length = MAX_DIGITS;
+		int result_length = length;
+		if (!(i == 2)) {
+			print_big_int(&input_number, 'm', 0);
+		}
+		else{
+			printf("%s", input);
+		}
+
+		printf(" * %d = ",i);
+		if (input_number.the_int[0] * i >= 10 && result_length == length ) {
+			result_length++;
+		}
+		struct BigInt result = {result_length};
+		multiply(&input_number, i, &result);
+		print_big_int(&result, 'm', 0);
+		copy_big_int(&result, &input_number);
+		printf("\n");
+
+		if (i == 9) {
+			print_big_int(&input_number, 'm', 0);
+		}
+	}
+
+	int counter = 1;
+
+	for (int i = 2; i <= 9; i++) {
+		length = MAX_DIGITS;
+		int result_length = length;
+		if (!(i == 2)) {
+			print_big_int(&input_number, 'd', counter - 1);
+		}
+
+		printf(" / %d = ",i);
+		if (input_number.the_int[0] * i >= 10 && result_length == length ) {
+			result_length++;
+		}
+		struct BigInt result = {result_length};
+		divide(&input_number, i, &result);
+		print_big_int(&result, 'd', counter);
+		copy_big_int(&result, &input_number);
+		printf("\n");
+
+		counter++;
+	}
+
+	return 0;
+}
+
+int strtobig_int(const char *str, int length, struct BigInt *big_int){
+
+	for (size_t i = 0; i < length; i++) {
```

Comment:
How will you handle carry overflows to add a new digit if the digit count increases?